File: Ahorn/entities/TimedStrawberry.jl
module XaphanHelperTimedStrawberry

using ..Ahorn, Maple

@mapdef Entity "XaphanHelper/TimedStrawberry" TimedStrawberry(x::Integer, y::Integer, keepEvenIfTimerRunOut::Bool=false, order::Integer=-1, checkpointID::Integer=-1, moon::Bool=false)

const placements = Ahorn.PlacementDict(
    "Timed Strawberry (Xaphan Helper)" => Ahorn.EntityPlacement(
        TimedStrawberry
    )
)

function Ahorn.selection(entity::TimedStrawberry)
    x, y = Ahorn.position(entity)
    return Ahorn.getSpriteRectangle("collectables/strawberry/normal00", x, y)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::TimedStrawberry, room::Maple.Room) = Ahorn.drawSprite(ctx, "collectables/strawberry/normal00", 0, 0)

end

File: Loenn/entities/TimedStrawberry.lua
local TimedStrawberry = {}

TimedStrawberry.name = "XaphanHelper/TimedStrawberry"
TimedStrawberry.depth = 0
TimedStrawberry.fieldInformation = {
    order = {
        fieldType = "integer",
    },
    checkpointID = {
        fieldType = "integer",
    }
}
TimedStrawberry.placements = {
    name = "TimedStrawberry",
    data = {
        keepEvenIfTimerRunOut = false,
        order = -1,
        checkpointID = -1,
        moon = false
    }
}

TimedStrawberry.texture = "collectables/strawberry/normal00"

return TimedStrawberry